rclrs/src/lib.rs

```diff
@@ -87,80 +88,26 @@ pub fn spin_once(node: &Node, timeout: i64) -> Result<(), RclError> {
 
     let context = &mut *node.context.get_mut();
 
-    unsafe {
-        rcl_wait_set_init(
-            &mut wait_set_handle as *mut _,
-            number_of_subscriptions,
-            number_of_guard_conditions,
-            number_of_timers,
-            number_of_clients,
-            number_of_services,
-            number_of_events,
-            context,
-            rcutils_get_default_allocator(),
-        ).ok()?;
-    }
+    let mut wait_set = WaitSet::new(
+        number_of_subscriptions,
+        number_of_guard_conditions,
+        number_of_timers,
+        number_of_clients,
+        number_of_services,
+        number_of_events,
+        context)?;
 
-    unsafe {
-        match rcl_wait_set_clear(&mut wait_set_handle as *mut _).ok() {
-            Ok(()) => (),
-            Err(rcl_err) => {
-                eprintln!("Unable to clear WaitSet! Error code: {:?}", rcl_err);
-                match rcl_wait_set_fini(&mut wait_set_handle as *mut _).ok() {
-                    Ok(()) => return Err(rcl_err),
-                    Err(rcl_err2) => {
-                        eprintln!("Trying to clear the WaitSet caused a second error!! Error code: {:?}", rcl_err2);
-                        return Err(rcl_err2);
-                    }
-                }
-            }
-        };
-    }
+    wait_set.clear()?;
```

Comment:
The entities in the `WaitSet` should already be initialized to `NULL` because the `new` function calls `rcl_get_zero_initialized_wait_set()`, so this call shouldn't be necessary.

Reply:
This was how it was set up originally, but I agree - it looks odd.